macros/src/lib.rs

```diff
@@ -60,10 +61,22 @@ use syn::{parse_quote, Error, Result};
 /// Furthermore, memoized functions cannot use destructuring patterns in their
 /// arguments.
 ///
+/// # Disabling memoization
+/// If you want to disable memoization for a function, you can use the `enabled`
+/// attribute to conditionally enable or disable memoization. This is useful for
+/// situations where small calls are more expensive than recomputing the result.
+/// This allows you to bypass hashing and caching while still dealing with the
+/// same function signature. And allows saving memory and time.
+///
+/// By default, all functions are memoized. To disable memoization, you must
+/// specify an `enabled = <expr>` attribute. The expression must evaluate to a
+/// boolean value. If the expression is `false`, the function will be executed
+/// without hashing and caching.
+///
 /// # Example
```

Comment:
I think I would prefer to move this example above the disabling part and remove `enabled` here and then add a shorter example for the enabled thing separately.